toplevel/role/role.go

```diff
@@ -303,3 +304,18 @@ func addOptionalOidcDefaults(instance string, roles []entry) {
 		}
 	}
 }
+
+func ValidateRoleUniqueness(desiredRoles []entry, toplevel string) error {
+	var uniqueNames = make(map[string]bool)
+	for _, role := range desiredRoles {
+		roleName := role.Name
+		uniqueRoleKey := strings.Join([]string{roleName, role.Mount}, "")
```

Comment:
```suggestion
		uniqueRoleKey := fmt.Sprintf("%s/%s", role.Mount, role.Name)
```

Reply:
This way we'll match vault cli's format of `auth_mount/role_name`

Reply:
I'm not sure why the `/` is necessary? The mount itself usually come with a slash

Reply:
I updated without the slash in the middle. Tested that it's not needed to result in auth_mount/role_name